core/src/main/java/pl/agh/edu/actor/component/rating/Rating.java

```diff
@@ -1,37 +1,56 @@
 package pl.agh.edu.actor.component.rating;
 
 import java.util.Arrays;
-import java.util.function.Function;
+import java.util.function.Consumer;
 import java.util.stream.IntStream;
 
-import com.badlogic.gdx.scenes.scene2d.ui.Image;
-import com.badlogic.gdx.scenes.scene2d.ui.Stack;
 import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.utils.NinePatchDrawable;
 
 import pl.agh.edu.actor.HotelSkin;
-import pl.agh.edu.config.GraphicConfig;
 
 public class Rating extends Table {
-
-	private final Table table = new Table();
-	private final Stack stack = new Stack();
 	private static final int maxRating = 5;
 	private final Star[] stars = IntStream.range(0, maxRating)
 			.mapToObj(index -> new Star(index, this))
 			.toArray(Star[]::new);
-	private int currentRating = 0;
-	private final Function<Integer, Void> function;
+	private int currentRating;
+	private final Consumer<Integer> function;
+
+	private boolean disabled = false;
 
-	public Rating(Function<Integer, Void> function) {
+	public Rating(Integer currentRating, Consumer<Integer> function) {
+		this.currentRating = currentRating;
 		this.function = function;
-		stack.add(new Image(HotelSkin.getInstance().getPatch("rating-background")));
-		stack.add(table);
-		Arrays.stream(stars).sequential().forEach(table::add);
-		this.add(stack).width(RatingStyle.getWidth()).height(RatingStyle.getHeight()).center();
+
+		this.setBackground(new NinePatchDrawable(HotelSkin.getInstance().getPatch("rating-background")));
+
+		Arrays.stream(stars).sequential().forEach(this::add);
+		this.setSize(RatingStyle.getWidth(), RatingStyle.getHeight());
```

Comment:
JESZCZE JEDNO NA KONIEC. to chyba nie ma wpływu na rozmiar tabeli. Jak do tej pory udawało mi się ustaliź rozmiar ustawiać przez to opakowanie w inną tabelę i ustawienie długości/wysokości przez add().width().height()

Reply:
The default behavior of Table is to size all children to their minimum size. You must remember that when you use Table, you no longer set the dimensions of the widget directly. You set the size of the cell, then the cell sizes the contents. Do not use setWidth() or setHeight().